packages/@misk/cli/src/utils/migrate.ts

```diff
@@ -3,46 +3,63 @@
  */
 
 import * as fs from "fs-extra"
-import { Files, IMiskTabJSON, JsonOptions } from "../utils"
+import {
+  Files,
+  IMiskTabJSON,
+  JsonOptions,
+  logDebug,
+  logFormatter,
+  remove,
+  path,
+  parseArgs
+} from "../utils"
 import { MiskVersion } from "./changelog"
 
-const moveOldBuildFile = async (filename: Files) => {
-  if (await fs.existsSync(filename)) {
-    fs.move(filename, `${Files.old}/${filename}`)
-  }
-}
+const tag = "migrate"
 
-const removeOldBuildFile = async (filename: Files) => {
-  if (await fs.existsSync(filename)) {
-    fs.remove(filename)
+const moveOldBuildFile = async (dir: string, filename: Files) => {
+  if (await fs.existsSync(path(dir, filename))) {
+    fs.move(path(dir, filename), path(dir, Files.old, filename))
   }
 }
 
-export const migrate = async () => {
-  console.log(
-    "[MIGRATE] Verify valid build files or migrate old build files to new miskweb generated build files"
-  )
-  let pkgMiskTab: IMiskTabJSON
-  if (await fs.existsSync(Files.package)) {
-    const pkg = await fs.readJson(Files.package)
-    if (pkg.name.startsWith("@misk/")) {
+export const migrateBuildFiles = (...args: any) => {
+  const { dir } = parseArgs(...args)
+  logDebug(tag, "", dir)
+  // Verify valid build files or migrate old build files to new miskweb generated build files
+  let pkgMiskTab: IMiskTabJSON = null
+
+  if (fs.existsSync(path(dir, Files.package))) {
+    let pkg = null
+    try {
+      pkg = fs.readJsonSync(path(dir, Files.package))
```

Comment:
we're still using exception based control flow here?

Reply:
synced offline. Ideally we'd use a "isValidJson" API, but for now we'll document why we're using exception-based control flow (we're using it because JSON can get partially written out due to flakiness around node FS APIs).